packages/devtools_app/lib/src/extensions/extension_service.dart

```diff
@@ -162,11 +162,19 @@ Future<String?> _connectedAppRootPath() async {
   // server and having the server look for the package folder that contains the
   // `.dart_tool` directory.
 
-  // Assume that the parent folder of `lib` is the package root.
+  // Assume that the parent folder of `lib` or `integration_test` is the package
+  // root.
   final libDirectoryRegExp = RegExp(r'\/lib\/[^\/.]*.dart');
   final libDirectoryIndex = fileUri.indexOf(libDirectoryRegExp);
+  final integrationTestDirectoryRegExp =
+      RegExp(r'\/integration_test\/[^\/.]*.dart');
+  final integrationTestDirectoryIndex =
+      fileUri.indexOf(integrationTestDirectoryRegExp);
```

Comment:
Maybe this could be a single regex, something like:

```dart
RegExp(r'\/(lib|integration_test|test|bin)\/[^\/.]*.dart');
```

Reply:
let's see